source/is-equal.d.ts

```diff
@@ -0,0 +1,21 @@
+/**
+Returns a boolean for whether the two given types are equal.
+
+@link https://github.com/microsoft/TypeScript/issues/27024#issuecomment-421529650
+@link https://stackoverflow.com/questions/68961864/how-does-the-equals-work-in-typescript/68963796#68963796
+
+@example
+```
+import type {IsEqual} from 'type-fest';
+
+type Expect<T extends true> = T
+type Case = Expect<IsEqual<Uppercase<"sample">, "SAMPLE">>
```

Comment:
```suggestion
type Expect<T extends true> = T;
type Case = Expect<IsEqual<Uppercase<'sample'>, 'SAMPLE'>>;
```

Reply:
I feel like the example is incomplete without also showing how `Case` is used.

Reply:
Thank you for your review. May I ask your opinion on which one is a good sample ?

A is the example I'm using. It's actually a little more complicated. The purpose is to check if the processed type is as expected.

```ts
// A
type Expect<T extends true> = T;
const ExpectTrue:Expect<IsEqual<Uppercase<'sample'>, 'SAMPLE'>> = true
```

B is a sample for many. I think Equal is generally used in types.

```ts
// B
type Includes<Value extends readonly any[], Item> =
	Value extends readonly [Value[0], ...infer rest]
		? IsEqual<Value[0], Item> extends true
			? true
			: Includes<rest, Item>
		: false;
```

Reply:
👍